### ISSUE TITLE
No qualifying bean of type org.springframework.cloud.config.client.ConfigClientProperties when using a parent application context

### ISSUE THREAD

**Issue by cbeattie-miovision** [spring-cloud/spring-cloud-config#179]
In Spring Cloud Config Client v1.0.1 there appears to be an issue when using Spring Cloud Config's auto configuration. Consider the following sample application:

```java
@Configuration
public class ParentApplication {
    public static void main(String[] args) {
        SpringApplicationBuilder parentApplicationBuilder = new SpringApplicationBuilder(ParentApplication.class);

        parentApplicationBuilder.run(args);

        parentApplicationBuilder.child(ChildApplication.class).run(args);
    }
}
```
```java
@SpringBootApplication
public class ChildApplication {
    public static void main(String[] args) {
        new SpringApplicationBuilder(ChildApplication.class).run(args);
    }
}
```

The following exception will be thrown:
```
Caused by: org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type [org.springframework.cloud.config.client.ConfigClientProperties] is defined
	at org.springframework.beans.factory.BeanFactoryUtils.uniqueBean(BeanFactoryUtils.java:432)
	at org.springframework.beans.factory.BeanFactoryUtils.beanOfType(BeanFactoryUtils.java:379)
	at org.springframework.cloud.config.client.ConfigClientAutoConfiguration.configClientProperties(ConfigClientAutoConfiguration.java:46)
```

The issue appears to be with this method in `org.springframework.cloud.config.client.ConfigClientAutoConfiguration`:
```
	@Bean
	public ConfigClientProperties configClientProperties(Environment environment,
			ApplicationContext context) {
		if (context.getParent() != null
				&& BeanFactoryUtils.beanNamesForTypeIncludingAncestors(
						context.getParent(), ConfigClientProperties.class).length > 0) {
			return BeanFactoryUtils.beanOfType(context.getParent(),
					ConfigClientProperties.class);
		}
		ConfigClientProperties client = new ConfigClientProperties(environment);
		return client;
	}
```

The parent context is non-null and `beanNamesForTypeIncludingAncesorts()` returns an array of one item, namely `configClientProperties` (which all seems correct). However when it attempts to load the bean it throws a `NoSuchBeanDefinitionException`.

**Comment by dsyer** [comment]
Fixed by #191

**Comment by cbeattie-miovision** [comment]
Thanks @dsyer.
